examples/dixonsparseelim.C

```diff
@@ -0,0 +1,214 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+/**\file examples/dixonsparseelim.C
+ @example examples/dixonsparseelim.C
+ @author Jean-Guillaume.Dumas@univ-grenoble-alpes.fr
+ * \brief Dixon System Solving Lifting using sparse LU
+ *        (direct match to dixendenseLU)
+ * \ingroup examples
+ */
+#include <iostream>
+#include <omp.h>
+#include "linbox/matrix/sparse-matrix.h"
+#include "linbox/solutions/solve.h"
+#include "linbox/util/matrix-stream.h"
+#include "linbox/solutions/methods.h"
+#include "linbox/util/error.h"
+
+using namespace LinBox;
+typedef Givaro::ZRing<Givaro::Integer> Ints;
+typedef DenseVector<Ints> ZVector;
+
+
+struct FixPrime {
+    typedef Givaro::Integer Prime_Type;
+    const Prime_Type _myprime;
+    FixPrime(const Givaro::Integer& i) : _myprime(i) {}
+    inline FixPrime &operator ++ () { return *this; }
+    const Prime_Type &operator * () const { return randomPrime(); }
+    const Prime_Type & randomPrime() const { return _myprime; }
+    void setBits(uint64_t bits) {}
+    template<class _ModField> void setBitsField() { }
+};    
+
+
+int main (int argc, char **argv) {
+        // Usage
+    if (argc < 2 || argc > 4) {
+        std::cerr << "Usage: solve <matrix-file-in-supported-format> [<dense-vector-file>]" << std::endl;
+        return 0;
+    }
+
+        // File
+    std::ifstream input (argv[1]);
+    if (!input) { std::cerr << "Error opening matrix file " << argv[1] << std::endl; return -1; }
+
+    
+    std::ifstream invect;
+    bool createB = false;
+    if (argc == 2) {
+        createB = true;
+    }
+    if (argc == 3) {
+        invect.open (argv[2], std::ifstream::in);
+        if (!invect) { 
+            createB = true;
+        } else {
+            createB = false;
+        }
+    }       
+    
+        // Read Integral matrix from File
+    Ints ZZ;
+    MatrixStream< Ints > ms( ZZ, input );
+    SparseMatrix<Ints> A (ms);
+    Ints::Element d;
+    std::cout << "A is " << A.rowdim() << " by " << A.coldim() << std::endl;
+    
+    {
+            // Print Matrix
+        
+            // Matrix Market
+            // std::cout << "A is " << A << std::endl;
+        
+            // Maple
+        if ( (A.rowdim() < 100) && (A.coldim() < 100) )
+            A.write(std::cout << "Pretty A is ", Tag::FileFormat::Maple) << std::endl;
+    }
+    
+        // Vectors
+    ZVector X(ZZ, A.coldim()),B(ZZ, A.rowdim());
+    ZVector U(ZZ, A.coldim());
+    
+    // B = A U
+    if (createB) {
+        std::cerr << "Creating a random {-1,1} vector " << std::endl;
+        srand48( BaseTimer::seed() );
+        for(ZVector::iterator it=B.begin();
+            it != B.end(); ++it)
+            if (drand48() <0.5)
```

Comment:
I thought our clarification of the randiter interface would have made possible to write somthing like 
```
ZRing<Integer>::RandIter G(2)
B.random(G);
``` 
but it seems that we do not offer uniform sampling  in intervals not starting in 0 in Givaro. Too bad.

Reply:
Now (with bool instead of Integer) in PR  linbox-team/givaro#196